src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -421,7 +421,8 @@ private void BuildLinkTag(string href, TagHelperAttributeList attributes, TagHel
                 }
                 else
                 {
-                    AppendAttribute(attribute.Name, attribute.Value, builder);
+                    attribute.CopyTo(builder);
+                    builder.AppendHtml(" ");
```

Comment:
Problem with this approach is if there's 0 attributes we still call into `AppendVersionedHref` on 432. That depends on a space after the tag name. The spacing has to be per-attribute write :cry: